test/reach.cc

```diff
@@ -0,0 +1,138 @@
+      },
+      "thor":{"logging":{"long_request": 100}},
+      "odin":{"logging":{"long_request": 100}},
+      "skadi":{"actons":["height"],"logging":{"long_request": 5}},
+      "meili":{"customizable": ["turn_penalty_factor","max_route_distance_factor","max_route_time_factor","search_radius"],
+              "mode":"auto","grid":{"cache_size":100240,"size":500},
+              "default":{"beta":3,"breakage_distance":2000,"geometry":false,"gps_accuracy":5.0,"interpolation_distance":10,
+              "max_route_distance_factor":5,"max_route_time_factor":5,"max_search_radius":200,"route":true,
+              "search_radius":15.0,"sigma_z":4.07,"turn_penalty_factor":200}},
+      "service_limits": {
+        "auto": {"max_distance": 5000000.0, "max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "auto_shorter": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "bicycle": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "bus": {"max_distance": 5000000.0,"max_locations": 50,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "hov": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50},
+        "isochrone": {"max_contours": 4,"max_distance": 25000.0,"max_locations": 1,"max_time": 120},
+        "max_avoid_locations": 50,"max_radius": 200,"max_reachability": 100,"max_alternates":2,
+        "multimodal": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 0.0,"max_matrix_locations": 0},
+        "pedestrian": {"max_distance": 250000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50,"max_transit_walking_distance": 10000,"min_transit_walking_distance": 1},
+        "skadi": {"max_shape": 750000,"min_resample": 10.0},
+        "trace": {"max_distance": 200000.0,"max_gps_accuracy": 100.0,"max_search_radius": 100,"max_shape": 16000,"max_best_paths":4,"max_best_paths_shape":100},
+        "transit": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+        "truck": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50}
+      }
+    })";
+  boost::property_tree::ptree conf;
+  rapidjson::read_json(ss, conf);
+  return conf;
+}
+
+GraphId begin_node(GraphReader& reader, const DirectedEdge* edge) {
+  // grab the node
+  const auto* tile = reader.GetGraphTile(edge->endnode());
+  const auto* node = tile->node(edge->endnode());
+  // grab the opp edges end node
+  const auto* opp_edge = tile->directededge(node->edge_index() + edge->opp_index());
+  return opp_edge->endnode();
+}
+
+void check_all_reach() {
```

Comment:
this is a new test i added which scans over all edges in utrecht and checks that the reaches are not wrong. there could be more tests in here to be honest but the degenerate cases are covered